### PR TITLE
Add Snap backend

### PR DESCRIPTION
**Pull request by watzon**
This PR adds support for the Snap package manager.

Changes:
- Added snap backend implementation
- Added snap to supported backends in README
- Added snap example in group files documentation

The implementation follows the same pattern as other backends, particularly taking inspiration from the apt backend structure.

### CODE REVIEW

File: src/backends/snap.rs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+        Ok(())
+    }
+
+    fn remove_packages(packages: &BTreeSet<String>, no_confirm: bool, _: &Config) -> Result<()> {
+        if !packages.is_empty() {
+            run_command(
+                ["snap", "remove"]
+                    .into_iter()
+                    .chain(Some("--assume-yes").filter(|_| no_confirm))
```
Comment (ripytide) [inline src/backends/snap.rs]: I get an error when using the `--assume-yes` option when I installed `snap` on my arch machine, are you sure this is a real option? I can't find it in the man page or on `snap install -h`.

`error: unknown flag 'assume-yes'`

**Comment by watzon** [comment]
Ahh you're right, that's an apt-get thing. I'll update it.

**Comment by watzon** [comment]
This should be good to go now

**Comment by ripytide** [comment]
Thanks!

**Comment by ripytide** [comment]
Released as `v0.2.5` :tada:

**Pull request merged**